spring-boot-project/spring-boot/src/main/java/org/springframework/boot/builder/SpringApplicationBuilder.java

```diff
@@ -109,6 +113,19 @@ protected SpringApplication createSpringApplication(Class<?>... sources) {
 		return new SpringApplication(sources);
 	}
 
+	/**
+	 * Creates a new {@link org.springframework.boot.SpringApplication} instances from the
+	 * given sources. Subclasses may override in order to provide a custom subclass of
+	 * {@link org.springframework.boot.SpringApplication}
+	 * @param resourceLoader the resource loader
+	 * @param sources the sources
+	 * @return the {@link org.springframework.boot.SpringApplication} instance
+	 * @since 1.1.0
+	 */
+	protected SpringApplication createSpringApplication(ResourceLoader resourceLoader, Class<?>... sources) {
```

Comment:
If we decided to pursue this, we should deprecate the other protected `createSpringApplication` in favor of this one.

Reply:
Fixed